tests/functional/alfven_wave/alfven_wave1d.py

```diff
@@ -118,54 +123,95 @@ def vthz(x):
 
 
 
-def plot(bhier):
-    times = np.sort(np.asarray(list(bhier.time_hier.keys())))
-
-    components  =("B_x", "B_y", "B_z")
-    ylims = ((0,1.5),(-0.25,0.25),(-0.25,0.25))
+####################################################################
+#                      post processing
+####################################################################
 
-    for component,ylim in zip(components,ylims):
-        for it,t in enumerate(times):
-            fig,ax = plt.subplots(figsize=(10,6))
-            for il,level in bhier.levels(t).items():
-                patches = level.patches
-                if il == 0:
-                    marker="+"
-                    alpha=1
-                    ls='-'
-                else:
-                    marker='o'
-                    alpha=0.4
-                    ls='none'
+def wave(x, a0, k, phi):
+    return a0*np.cos(k*x + phi)
 
-                for ip, patch in enumerate(patches):
-                    val   = patch.patch_datas["EM_"+component].dataset[:]
-                    x_val = patch.patch_datas["EM_"+component].x
-                    label="${}$ level {} patch {}".format(component,il,ip)
-                    ax.plot(x_val, val, label=label,
-                            marker=marker, alpha=alpha, ls=ls)
-                    ax.set_ylim(ylim)
 
-            ax.legend(ncol=4)
-            ax.set_title("t = {:05.2f}".format(t))
-            fig.savefig("{}_{:04d}.png".format(component,it))
-            plt.close(fig)
+def phase_speed(run_path, ampl, xmax):
+    from scipy.signal import medfilt
+    import os
+    time = np.asarray(get_times(os.path.join(run_path,"EM_B.h5")))
+    r = Run(run_path)
+    phase = np.zeros_like(time)
+    amplitude = np.zeros_like(time)
+    wave_vec = np.zeros_like(time)
 
+    for it, t in enumerate(time):
+        B = r.GetB(t)
+        by, xby = finest_field(B, "By")
+        a, k, phi = curve_fit(wave, xby, by,p0=(ampl, 2*np.pi/xmax, 0))[0]
+        phase[it] = phi
+        amplitude[it] = a
+        wave_vec[it]  = k
 
+    vphi = medfilt(np.gradient(phase, time)/wave_vec, kernel_size=7)
+    return vphi, time, phase, amplitude, wave_vec
 
 
+def get_times(path):
+    import h5py
+    f = h5py.File(path, 'r')
+    times = np.array(sorted([float(s.strip("t")) for s in list(f.keys())]))
```

Comment:
https://github.com/PHAREHUB/PHARE/pull/494 will break this, maybe refactor to hierarchy.py as

"times_from_h5_file(path):" / "h5_file_times(path)"

can dedupe conserv.py too